### PR TITLE
[#6504] Make AVU-by-condInput association atomic (main)

### PR DESCRIPTION
**Pull request by alanking**
Core tests are running now

The latest commit on this branch includes some renames and comments. I'm not sure how helpful these changes are so I decided not to squash them yet until I got some feedback.

### CODE REVIEW

**Comment by alanking** [comment]
Experiencing intermittent failures with core tests which seem unrelated to my changes, so I will need to have a closer look at those

File: server/core/src/finalize_utilities.cpp [hunk 1]
```diff
@@ -34,24 +36,44 @@ namespace irods
...
+        // clang-format off
+        auto json_input = json{{"entity_name", _inp.objPath},
+                               {"entity_type", "data_object"},
+                               {"operations", json::array()}};
+        // clang-format on
+
+        // The strings in the avu_list are taken 3 at a time because METADATA_INCLUDED_KW
+        // must contain a value with a string of triplets of semicolon-separated tokens.
+        for (std::size_t i = 0; i + 2 < avu_list.size(); i += 3) {
+            json_input["operations"].push_back({{"operation", "add"},
```
Comment (korydraughn) [inline server/core/src/finalize_utilities.cpp:62]: `json_input["operations"]` can be moved out of the loop to avoid multiple lookups of "operations".

Comment (korydraughn) [reply server/core/src/finalize_utilities.cpp:62]: Alternatively, you can build up the json array directly and if all of that succeeds, create the json_input object afterwards.

Comment (alanking) [reply server/core/src/finalize_utilities.cpp:62]: Good idea, I'll try that.

Comment (alanking) [reply server/core/src/finalize_utilities.cpp]: Alternative implementation seems to work, so I'll resolve this.

**Comment by alanking** [comment]
[e5f25a2](https://github.com/irods/irods/pull/6508/commits/e5f25a2ecfc45a0a128bedaa0c3e6dcb7613f607) - Commit message has the numbers backwards. Should be 6504, not 6405.

**Comment by korydraughn** [comment]
Should we try running the tests again?

**Comment by alanking** [comment]
Still seeing failures when running tests on tip of main without these changes, so I need to get to the bottom of that

**Comment by alanking** [comment]
After running a couple of times, all of the tests passed

File: server/core/src/finalize_utilities.cpp [hunk 2]
```diff
@@ -34,24 +36,43 @@ namespace irods
...
+                                  {"units", avu_list[i + 2]}});
+        }
+
+        // clang-format off
+        const auto json_input_str = json{{"entity_name", _inp.objPath},
+                                         {"entity_type", "data_object"},
+                                         {"operations", operations}}.dump();
+        // clang-format on
+
+        server_log::debug("[{}:{}] - JSON input: [{}]", __func__, __LINE__, json_input_str);
```
Comment (trel) [inline server/core/src/finalize_utilities.cpp:69]: i thought i had heard the new logger automatically emits func and line?

**Comment by trel** [comment]
good to squash and swap issue number?

File: server/core/src/finalize_utilities.cpp (hunk 2 above)
Comment (alanking) [reply server/core/src/finalize_utilities.cpp:70]: It does not seem to be automatic based on the other log output I am seeing, but perhaps there is an option I don't know about, though.

**Comment by alanking** [comment]
Squashed and corrected issue number

**Comment by trel** [comment]
last line of last commit message...

"allwos us to sidestep" -> "allows us to sidestep"

**Comment by alanking** [comment]
Done

**Review by trel — APPROVED** [review]

**Pull request merged**